shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1003,22 +1003,13 @@ void WebContents::RenderFrameCreated(
     rwh_impl->disable_hidden_ = !background_throttling_;
 }
 
-void WebContents::RenderViewHostChanged(content::RenderViewHost* old_host,
-                                        content::RenderViewHost* new_host) {
-  currently_committed_process_id_ = new_host->GetProcess()->GetID();
-}
-
 void WebContents::RenderViewDeleted(content::RenderViewHost* render_view_host) {
   // This event is necessary for tracking any states with respect to
   // intermediate render view hosts aka speculative render view hosts. Currently
   // used by object-registry.js to ref count remote objects.
   Emit("render-view-deleted", render_view_host->GetProcess()->GetID());
 
-  if (-1 == currently_committed_process_id_ ||
-      render_view_host->GetProcess()->GetID() ==
-          currently_committed_process_id_) {
-    currently_committed_process_id_ = -1;
-
+  if (web_contents()->GetRenderViewHost() == render_view_host) {
```

Comment:
Does this mean `RenderViewHostChanged` is now emitted for speculative hosts ?

Reply:
You mean `current-render-view-deleted`? If you do - quite the contrary. When we do a call to window.open in child window and there is a redirection from http to https speculative rvh is being destroyed - and that is expected. As there was no swap of rvh `currently_committed_process_id_` is set to `-1` so we generate `current-render-view-deleted` and destroy browser window. With this change it won't happen as rvh is assigned to web contents only when it is commited and it won't happen for speculative rvh.

Reply:
ah makes sense, thanks for the details!